### DISCUSSION TITLE
Issue with Copilot Plugin Causing Jetbrains Idea Window to Freeze

### DISCUSSION THREAD

**Discussion by imztone** [community/community#75695] [Copilot Conversations]
**Description**:
Copilot 1.4.0.3748 JetBrains Idea 2023.2.5
The Copilot plugin appears to be stuck in a login loop with Copilot Chat whenever activated in JetBrains idea, leading to a frozen state of the entire application window.

**Steps to reproduce**:
I have discovered that uninstalling the Copilot plugin resolves the freezing issue. However, upon reinstallation, the problem consistently reoccurs, indicating a direct link between the plugin's activity and the freezing issue.
Specifically, the plugin consistently attempts to log in to Copilot Chat, resulting in the freezing of the IDEA window. 

Please find attached the thread dump and error logs for a detailed understanding of the problem.
[threadDump-20231116-093442.txt](https://github.com/community/community/files/13374034/threadDump-20231116-093442.txt)

**Comment by imztone** [comment]
got an email reply

**Reply by zhujiefirst** [reply]
how to resolve it,  I have this problem, too

**Reply by imztone** [reply]
Sorry for missing the notification earlier. Ran into an error because of bad network conditions, which messed up my login. If you're in mainland China, you might want to try using tun mode to proxy the Copilot plugin. This trick seems to do the job. 
Once you're logged in and it's cached, the problem doesn't come back.

Also, I reached out to a Copilot team member about this but didn’t get a fix for the IDE freezing when you can't log in.

**Just a heads-up, I’ve actually stopped using Copilot recently, so the info I'm sharing might be a bit dated.**

**Comment by CodeWithKyrian** [comment]
I've got the same issue here on Mac. Any update on this please?

**Comment by ramAdam** [comment]
same issue on linux, any updates ?

**Comment by wuzejinCN** [comment]
Copilot prevent IntelliJ IDEA to close when some process  is attached. IDEA 2023.2, Copilot 233.13135.116.
![Uploading image.png…]()

**Comment by nathand8** [comment]
I had the same issue. It went away when I stopped using VPN.

If you're having problems, maybe try switching networks. Post here if that works (to assist with debugging). A plugin shouldn't crash an app if there is a network problem 😆

**Reply by adriancs1** [reply]
The latest plugin v 1.5.20.6554 doesn't work with Rider 2024.2 . I keep getting prompted to log in via GitHub and while the UI shows that I successfully log in, the logs shows INFO - #copilot - [auth] Failed to get copilot token due to 401 status. Please sign out and try again. And GitHub Chat UI shows the never ending messages : Initializing Checking Copilot Chat prerequisites.

To get GitHub Copilot working again with Rider2024.2, I had to uninstall v 1.5.20.6554 (which is the latest version as of this writing) and install the previous working version v. 1.5.19.6527. I hope that this helps others that ran into the same issue.

**Comment by assafos** [comment]
Happens to me as well. This is super annoying!!!

Intellij ver 2024.3.1.1 (and several older versions)
Copilot ver 1.5.30.8450-242-nightly (and also several stable versions before)

1. When opening copilot tab for the first time (or when closing and opening the tab) - it takes up to 1 min to load and freezes intellij
2. When i'm typing in the tab chatbox - i have a delay every couple of seconds. Something really annoying.
3. Intellij stutters when chat is open. Every 1-2 seconds everything gets stuck - for instance if i scroll with the mouse, it stutters.
4. When opening the shitty chat window on a different window - only the chat stutters, and the IDE works fine.
5. Huge amounts of cpu usage (>150%), didn't check the memory - but I wouldn't be surprised if I see some nasty stuff there

WHAT THE HELL??

I tried restarting intellij, invalidate cache, etc. Also removed plugins and increased intellij mem - nothing helped.
Why the hell is this thread closed without answers??

**Comment by friederbluemle** [comment]
Pretty crazy that a malfunctioning plugin can permanently render the entire UI frozen and unusable. What's similarly surprising is that this issue was first reported more than a year ago and I just ran into it. I was using the latest version. Why is this issue closed?
I had to manually go into WebStorm's plugin directory on the file system and `rm -rf github-copilot-intellij` just to be able to use my IDE again (after force closing it). Disappointing.

**Comment by CelsoOliveiraTR** [comment]
This is issue is being pretty disrupting and slowing down development. Can anyone give some feedback why it is happening and what is being done to fix it? I noticed that it is worse for the projects I have more copilot usage.

**Comment by assafos** [comment]
*** Update + solution for lagging UI with copilot ***

Apparently this is a classic case of horrible UX. 
It seems that we're just using the SAME old context every time, without opening a new one.
Why? Because we simply didn't know we need to open a new one every once in a while, using the "+" icon on top.

Not opening a new context makes the plugin send the whole conversation history from day 0 every time the copilot plugin is loaded, hence the annoying UI freeze / lags that worsen over time. This is where it differs from chatGPT, where the whole context is being saved and fetched quickly from the cloud and not being sent from your IDE environment every time like in copilot. Copilot does that for security reasons.

So why didn't the creators of copilot add a simple popup to teach us about that "+" button or added a simple tutorial on first load? 
Who knows. I have 10 people in my team, only 30% of them knew about the existence of this button.

Assaf

**Reply by CelsoOliveiraTR** [reply]
Thank you! It was very helpful. I notice that creating a new conversation helped. I also notice when I click on the conversation history and try to click on any conversation I can see that the IDE struggles to open those historical conversation. I hope they soon address that too. For now the "+" button will help for sure.

**Reply by wwzeng1** [reply]
Thanks for the find! I am building [Sweep AI](https://docs.sweep.dev) and we put in-product tutorials like "click '+' to start a new chat" and it's helped a lot.

**Comment by harrisbaldon** [comment]
Mine got fixed by uninstalling intellij and then deleted the configuration files.
Installed again via homebrew.
https://formulae.brew.sh/cask/intellij-idea

**Reply by kaustubhsuryakantdeshpande** [reply]
In my case happening with the Android Studio

**Reply by wildsky95** [reply]
I fixed it by deleteing every configuration that intelij has for defferent versions. and remove the .idea from the project directory.  And after this when you are loading it for the first time dont load any configuration and skip.

**Reply by kaustubhsuryakantdeshpande** [reply]
Tried it not working

**Reply by cgorrieri** [reply]
I deleted the the file `copilotDiffState.xml` in the `.idea` folder of my project and that solved the freeze.

**Reply by kishmakov** [reply]
Thanks @cgorrieri : deleting `copilotDiffState.xml` worked for me (for CLion).

**Comment by vinaykrajput** [comment]
I figured out,

its problem with themes or something which is not configured properly as layout rendering

so whenever you face this issue, try changing theme  (windows shortcut => Ctrl + ` ), the moment i changed theme Github Copilot chat window became responsive,

Try it and let me know if this resolved Github copilot crashing issue with intellij
